tt_metal/impl/device/device.hpp

```diff
@@ -117,6 +117,12 @@ class Device {
         return tt::Cluster::instance().get_connected_ethernet_core(std::make_tuple(this->id_, eth_core));
     }
 
+    CoreCoord get_and_configure_corresponding_eth_core(
+        CoreCoord physical_dispatch_core, EthRouterMode mode, chip_id_t connected_chip_id) const {
+        return tt::Cluster::instance().get_and_configure_corresponding_eth_core_for_device(
```

Comment:
most of our apis accept logical core coordinates. also should `CoreCoord physical_dispatch_core` be `tt_cxy_pair logical_dispatch_location` and then get the connected chip based off the chip in `logical_dispatch_location`?

Reply:
also should we break these up into two sep apis, one to get the eth core and another to configure it?

Reply:
I can change the core coord to be logical and group the connected chip to `tt_cxy_pair`. I think breaking this to two apis makes sense, since I guess we might want to call the getter in other places.